src/order/lattice.lean

```diff
@@ -139,9 +142,14 @@ begin
   cases A; cases B; injection this; congr'
 end
 
-lemma directed_of_sup {β : Type*} {r : β → β → Prop} {f : α → β}
-  (hf : ∀a₁ a₂, a₁ ≤ a₂ → r (f a₁) (f a₂)) : directed r f :=
-assume x y, ⟨x ⊔ y, hf _ _ le_sup_left, hf _ _ le_sup_right⟩
+lemma directed_of_sup {β : Type*} {r : rel β β} {f : α → β}
+  (hf : ((≤) ⟹ r).diag f) : directed r f :=
```

Comment:
Again, someone who doesn't know how to read `((≤) ⟹ r).diag f` won't be able to tell whether this lemma is useful to them.

Reply:
Removed `diag`. I'd prefer to leave `⇒`. I'll add a comment tonight.